fingerpaintview/src/main/java/tech/picnic/fingerpaintview/FingerPaintImageView.kt

```diff
@@ -164,14 +188,54 @@ class FingerPaintImageView @JvmOverloads constructor(context: Context,
 
     override fun onDraw(canvas: Canvas?) {
         super.onDraw(canvas)
-        paths.forEach { canvas?.drawPath(it.first, it.second) }
+        mBitmap = Bitmap.createBitmap(widthBitmap,
+                heightBitmap,
+                Bitmap.Config.ARGB_8888)
+        mCanvas = Canvas(mBitmap)
+
+        canvas?.save()
+        for (index in paths.indices){
+            if(index >= countDrawn) {
+                if (isEmboos) paths[index].second.maskFilter = mEmboss
+                else if (isBlur) paths[index].second.maskFilter = mBlur
+                else if (!isBlur && !isEmboos) paths[index].second.maskFilter = null
+            }
+            mCanvas?.drawPath(paths[index].first, paths[index].second)
+        }
+        canvas?.drawBitmap(mBitmap,0f,0f,mBitmapPaint)
+        canvas?.restore()
+    }
+
+    /**
+     * Enable normal mode
+     */
+    fun normal() {
+        isEmboos = false
```

Comment:
These set of booleans (`isBlur`, `normal` and `isEmboos`) are representing the drawing mode which could be modeled as an enum. That would make the code much more simpler and maintainable if we want to add even more modes :)